packages/zendesk-adapter/src/filters/guide_order/categories_order.ts

```diff
@@ -0,0 +1,88 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Change, Element, getChangeData,
+  InstanceElement,
+  isInstanceElement, ReferenceExpression,
+} from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { FilterCreator } from '../../filter'
+import { BRAND_TYPE_NAME, CATEGORY_TYPE_NAME } from '../../constants'
+import {
+  CATEGORIES_FIELD,
+  CATEGORIES_ORDER,
+  createOrderInstance,
+  createOrderType,
+  deployOrderChanges,
+} from './guide_orders_utils'
+import { FETCH_CONFIG } from '../../config'
+
+/**
+ * Handle the order of categories in brand
+ */
+const filterCreator: FilterCreator = ({ client, config }) => ({
+  /** Create an InstanceElement of the categories order inside the brands */
+  onFetch: async (elements: Element[]) => {
+    const categories = elements.filter(isInstanceElement)
+      .filter(e => e.elemID.typeName === CATEGORY_TYPE_NAME)
+    const brands = elements.filter(isInstanceElement)
+      .filter(e => e.elemID.typeName === BRAND_TYPE_NAME)
+
+    // If Guide is not enabled in Salto, we don't need to do anything
+    if (!config[FETCH_CONFIG].enableGuide) {
+      return
+    }
+
+    const orderType = createOrderType(CATEGORY_TYPE_NAME)
+    elements.push(orderType)
+
+    // If the brand doesn't have Guide activated, do nothing
+    brands.filter(b => b.value.has_help_center).forEach(brand => {
+      const orderInBrandElement = createOrderInstance({
+        parent: brand,
+        parentField: 'brand',
+        orderField: CATEGORIES_FIELD,
+        childrenElements: categories,
+        orderType,
+      })
+      brand.value.categories = new ReferenceExpression(
+        orderInBrandElement.elemID, orderInBrandElement
+      )
+      elements.push(orderInBrandElement)
+    })
```

Comment:
styling/convention nit:
```suggestion
    const categoryOrderInstances = brands
      .filter(b => b.value.has_help_center)
      .map(brand => {
        const categoryOrderInst = createOrderInstance({
          parent: brand,
          parentField: 'brand',
          orderField: CATEGORIES_FIELD,
          childrenElements: categories,
          orderType,
        })
        brand.value.categories = new ReferenceExpression(
          orderInBrandElement.elemID, orderInBrandElement
        )
        return categoryOrderInst
    })
    categoryOrderInstances.forEach(inst => elements.push(inst))
```

note that the reason for the `.forEach(push))` rather than `push(...categoryOrderInstances)` is that when there is a large number of items to push the process may crash due to js limitations